### PR TITLE
add elemText proc to xmltree

### PR DESCRIPTION
**Pull request by LaughingBubba**
`text*` proc  cannot be used with an xnElement type. 
`elemText*` is intended to fill that gap. 

This proc  gets the associated text for an element node `n`. Finds all the top level text nodes for an element
and returns a concatenated string of their contents. An optional string can be passed to be used as a text node separator. 

If there are no text nodes, an empty string is returned.

### CODE REVIEW

**Comment by ghost** [comment]
I didn't check the PR in detail, but how is this different from `xmltree.innerText` - https://nim-lang.org/docs/xmltree.html#innerText%2CXmlNode?

**Comment by LaughingBubba** [comment]
`innerText` does all nodes with an `fText` property and ripples through all the child nodes of a node.

`elemText` only does the top level `xnText`'s nodes for only an `xnElement`. AFAIK there is no equivalent proc in the `xmltree` library. 

Mr Rumpf didn't like my last effort but left instructions as to what would be acceptable. Check the closed PR [here](https://github.com/nim-lang/Nim/pull/20143)

**Comment by Varriount** [comment]
Just curious, do other language's XML libraries have an equivalent to this? If so, what is it called?

File: lib/pure/xmltree.nim [hunk 1]
```diff
@@ -296,6 +299,53 @@ proc innerText*(n: XmlNode): string =
...
+    assert x.elemText(" ") == "abc xyz"
+   
+  assert n.k == xnElement
+  
+  proc worker(res: var string, n: seq[XmlNode], i: int = 0) =
+    if i < n.len:
+      if n[i].k == xnText:
+        res.add(n[i].fText)
+        if i < n.len - 1:
+          res.add(separator)
```
Comment (Araq) [inline lib/pure/xmltree.nim]: The correct logic here is:

```suggestion
        if res.len > 0: res.add(separator)
        res.add(n[i].fText)
```

**Comment by Araq** [comment]
`plainText` or `stripTags` would be better names.

**Comment by bung87** [comment]
it's like `innerText` and `textContent` in html

**Comment by Araq** [comment]
If in doubt, leave it out. These things never exactly do what you need anyway and you're better off with custom logic instead.

**Pull request closed without merging**